Kickstarter-iOS/ViewModels/ChangePasswordViewModelTests.swift

```diff
@@ -0,0 +1,162 @@
+import Foundation
+import Prelude
+import Result
+import XCTest
+
+@testable import KsApi
+@testable import Kickstarter_Framework
+@testable import ReactiveExtensions_TestHelpers
+
+final class ChangePasswordViewModelTests: TestCase {
+  private let vm: ChangePasswordViewModelType = ChangePasswordViewModel()
+
+  private let activityIndicatorShouldShowObserver = TestObserver<Bool, NoError>()
+  private let changePasswordFailureObserver = TestObserver<String, NoError>()
+  private let changePasswordSuccessObserver = TestObserver<DiscoveryParams?, NoError>()
+  private let confirmNewPasswordBecomeFirstResponderObserver = TestObserver<Void, NoError>()
+  private let currentPasswordBecomeFirstResponder = TestObserver<Void, NoError>()
+  private let currentPasswordPrefillValueObserver = TestObserver<String, NoError>()
+  private let dismissKeyboardObserver = TestObserver<Void, NoError>()
+  private let newPasswordBecomeFirstResponderObserver = TestObserver<Void, NoError>()
+  private let onePasswordButtonIsHiddenObserver = TestObserver<Bool, NoError>()
+  private let onePasswordFindPasswordForURLStringObserver = TestObserver<String, NoError>()
+  private let saveButtonIsEnabledObserver = TestObserver<Bool, NoError>()
+  private let validationErrorLabelIsHiddenObserver = TestObserver<Bool, NoError>()
+  private let validationErrorLabelMessageObserver = TestObserver<String, NoError>()
+
+  override func setUp() {
+    super.setUp()
+
+    self.vm.outputs.activityIndicatorShouldShow.observe(activityIndicatorShouldShowObserver.observer)
+    self.vm.outputs.changePasswordFailure.observe(changePasswordFailureObserver.observer)
+    self.vm.outputs.changePasswordSuccess.observe(changePasswordSuccessObserver.observer)
+    self.vm.outputs.confirmNewPasswordBecomeFirstResponder
+      .observe(confirmNewPasswordBecomeFirstResponderObserver.observer)
+    self.vm.outputs.currentPasswordBecomeFirstResponder.observe(currentPasswordBecomeFirstResponder.observer)
+    self.vm.outputs.currentPasswordPrefillValue.observe(currentPasswordPrefillValueObserver.observer)
+    self.vm.outputs.dismissKeyboard.observe(dismissKeyboardObserver.observer)
+    self.vm.outputs.newPasswordBecomeFirstResponder.observe(newPasswordBecomeFirstResponderObserver.observer)
+    self.vm.outputs.onePasswordFindPasswordForURLString
+      .observe(onePasswordFindPasswordForURLStringObserver.observer)
+    self.vm.outputs.onePasswordButtonIsHidden.observe(onePasswordButtonIsHiddenObserver.observer)
+    self.vm.outputs.saveButtonIsEnabled.observe(saveButtonIsEnabledObserver.observer)
+    self.vm.outputs.validationErrorLabelIsHidden.observe(validationErrorLabelIsHiddenObserver.observer)
+    self.vm.outputs.validationErrorLabelMessage.observe(validationErrorLabelMessageObserver.observer)
+  }
+
+  func testChangePassword() {
+    let service = MockService()
+    let delay: DispatchTimeInterval = .milliseconds(500)
```

Comment:
Are you setting the delay to 500 milliseconds for any reason? If not, using `scheduler.advance()` should be enough. It's not a big deal though.

Reply:
Yep you're right all I needed was `scheduler.advance()` - thanks!